core/src/main/java/org/bitcoinj/base/utils/ByteUtils.java

```diff
@@ -356,4 +357,28 @@ public static boolean checkBitLE(byte[] data, int index) {
     public static void setBitLE(byte[] data, int index) {
         data[index >>> 3] |= bitMask[7 & index];
     }
+
+    /**
+     * Provides a byte array comparator.
+     * @return A comparator for byte[]
+     */
+    public static Comparator<byte[]> arrayUnsignedComparator() {
```

Comment:
I think this could use some tests?

Reply:
Agreed.

Reply:
(I do remember there were some pecularities about the ordering.)

Reply:
Tests added.